showcase/treemap/tree-tools.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Find branch in tree by omen pair - key-value
+ * @author Akiyamka <akiyamka@gmail.com>
+ *
+ * @param {Object} omen - how to find branch. Example - { id: 123456 }
+ * @param {Object} options [{children: 'children', returnChild: true}];
+ * @param {Boolean} options.returnChild - return whole branch or not
+ * if true return branch started from found child,
+ * @returns {Object} if options.returnChild true - return branch started from found child,
+ * else - return whole branch where child founded.
+ */
+export function findBranchByOmen(omen, options) {
+  const [[omenProp, omenValue]] = Object.entries(omen);
+  const opt = {
+    children: 'children',
+    returnChild: true,
+    ...options
+  };
+
+  function _findBranch(tree) {
+    if (omenValue === tree[omenProp]) {
+      return tree;
+    }
+
+    /* Needed for recursive reduce and find */
+    if (!tree[opt.children]) {
+      return false;
+    };
+
+    return opt.returnChild
+      ? tree[opt.children].reduce((result, branch) => _findBranch(branch) || result, false)
+      : tree[opt.children].find(branch => _findBranch(branch));
+  }
+
+  return data => _findBranch(data);
+}
+
+/**
+ * Fold childrens, and recalculate weights on leafs.
+ * Used to create tree slice of the required depth
+ * @author Akiyamka <akiyamka@gmail.com>
+ *
+ * @param {Object} tree
+ * @param {Object} options
+ * @param {String} options.children - the field where to take children
+ * @param {String} options.value - the field where the values will be written
+ * @param {String} options.maxLevel - the maximum depth of leaf
+ * @param {String} options.clearValues [false] - if true - values saved only on leafs
+ *
+ * @returns {Object} folded tree
+ */
+export function foldChilndrens(tree, options, level = 0) {
+  const opt = {
+    children: 'children',
+    value: 'value',
+    maxLevel: 2,
+    clearValues: false,
+    ...options
+  };
+
+  const calcLeafWeight = childrn => childrn.reduce(
```

Comment:
```suggestion
  const calcLeafWeight = children => children.reduce(
```

Reply:
changed in c2d79bbc144878ba8272a8955737e176599b9338